cmd/app/app.go

```diff
@@ -110,16 +110,28 @@ func NewCommand(ctx context.Context) *cobra.Command {
 				return fmt.Errorf("failed to setup image registry clients: %s", err)
 			}
 
-			c := controller.NewPodReconciler(opts.CacheTimeout,
+			_ = client
+
+			podController := controller.NewPodReconciler(opts.CacheTimeout,
 				metricsServer,
 				client,
 				mgr.GetClient(),
 				log,
 				opts.RequeueDuration,
 				opts.DefaultTestAll,
 			)
+			if err := podController.SetupWithManager(mgr); err != nil {
+				return err
+			}
 
-			if err := c.SetupWithManager(mgr); err != nil {
+			kubeController := controller.NewKubeReconciler(
```

Comment:
I was thinking, if it's worth checking ig `opts.KubeChannel` is not empty, and not starting the kubernetes channel version Reconciler... Maybe that's the SetupWithManager's responsibility? wdyt?